app/ts/components/subcomponents/ConfigureRpcConnection.tsx

```diff
@@ -0,0 +1,253 @@
+	return context
+}
+
+export const ConfigureRpcConnection = ({ rpcEntries, rpcInfo }: { rpcEntries: RpcEntries ,rpcInfo?: RpcEntry | undefined }) => {
+	const modalRef = useRef<HTMLDialogElement>(null)
+
+	const verifyInputsAndSave = (event: Event) => {
+		// TODO: current version preact don't ship with SubmitEvent type
+		if (!(event instanceof SubmitEvent)) return
+
+		// abort if dialog submitter was a cancel request
+		if (event.submitter instanceof HTMLButtonElement) {
+			if (event.submitter.value === 'cancel') {
+				// reset the form to it's previous state
+				if (event.target instanceof HTMLFormElement) event.target.reset()
+				modalRef.current?.close()
+				return
+			}
+
+			if (event.submitter.value === 'remove') {
+				// remove current rpc
+				sendPopupMessageToBackgroundPage({
+					method: 'popup_set_rpc_list',
+					data: rpcEntries.filter(entry => entry.httpsRpc !== rpcInfo?.httpsRpc)
+				})
+				return
+			}
+		}
+
+		// handle form validation and submission
+		if (event.target instanceof HTMLFormElement) {
+			const formData = new FormData(event.target)
+
+			const newRpcEntry = {
+				name: formData.get('name')?.toString() || '',
+				chainId: BigInt(formData.get('chainId')?.toString() || ''),
+				httpsRpc: formData.get('httpsRpc')?.toString() || '',
+				currencyName: formData.get('currencyName')?.toString() || '',
+				currencyTicker: formData.get('currencyTicker')?.toString() || '',
+				minimized: true,
+				primary: false,
+			} satisfies RpcEntry
+
+			const parsedRpcData = parseData(newRpcEntry)
+
+
+			if (!parsedRpcData.success) {
+				event.preventDefault()
+				console.error('RPC Info Parse', parsedRpcData)
+				return
+			}
+
+			sendPopupMessageToBackgroundPage({
+				method: 'popup_set_rpc_list',
+				data: rpcEntries.filter(entry => entry.httpsRpc !== parsedRpcData.value.httpsRpc).concat([parsedRpcData.value])
+			})
+		}
+	}
+
+	const showConfigurationModal = () => modalRef.current?.showModal()
+
+	return (
+		<ConfigureRpcProvider>
+			{ rpcInfo
+				? <button type = 'button' onClick = { showConfigurationModal } class = 'btn btn--outline'>Edit</button>
+				: <button type = 'button' onClick = { showConfigurationModal } class = 'btn btn--outline' style = 'border-style: dashed'>+ New RPC Connection</button>
+			}
+			<dialog class = 'dialog' ref = { modalRef } onSubmit = { verifyInputsAndSave }>
+				<ConfigureRpcForm defaultValues = { rpcInfo } />
+			</dialog>
+		</ConfigureRpcProvider>
+	)
+}
+
+const ConfigureRpcForm = ({ defaultValues }: { defaultValues?: RpcEntry }) => {
+	const confirmRemoval = useSignal(false)
+	const { rpcUrlQuery } = useConfigureRpc()
+
+	const chainIdDefault = useComputed(() => {
+		if (rpcUrlQuery.value.state === 'resolved') return BigInt(rpcUrlQuery.value.value.result).toString()
+		return defaultValues?.chainId?.toString() || ''
+	})
+
+	const networkNameDefault = useComputed(() => CHAIN_NAMES.get(chainIdDefault.value) || defaultValues?.name || '')
```

Comment:
did you mean "Chain Name" here?

Chain ID will be supplied by the RPC response or pre-defined from storage. The only time the field value is an empty string is when a user initially adds a new RPC server where the form will not submit anyways since it's not valid yet at that point.

Reply:
when there's no chain name available in our system, we need to come up with a name for the chain anyway